pastas/io/base.py

```diff
@@ -184,6 +127,128 @@ def _load_model(data: dict) -> Model:
+            "RechargeModel using a Linear recharge model. Make sure to save "
+            "this file first using Pastas version 0.22.0 as this file cannot be "
+            "loaded in newer Pastas versions. This will automatically update "
+            "your model to the newer RechargeModel stress model."
+        )
+        logger.error(msg=msg)
+        raise NotImplementedError(msg)
+
+    # TODO Deal with old parameter value b in HantushWellModel: b_new = np.log(b_old)
+    if (ts["stressmodel"] == "WellModel") and (
+        version.parse(data["file_info"]["pastas_version"]) < version.parse("0.22.0")
+    ):
+        logger.warning(
+            "The value of parameter 'b' in HantushWellModel was modified in 0.22.0: "
+            "b_new = log(b_old). The value of 'b' is automatically updated on load."
+        )
+        wnam = ts["name"]
+        for pcol in ["initial", "optimal", "pmin", "pmax"]:
+            if wnam + "_b" in data["parameters"].index:
+                if data["parameters"].loc[wnam + "_b", pcol] > 0:
+                    data["parameters"].loc[wnam + "_b", pcol] = log(
+                        data["parameters"].loc[wnam + "_b", pcol]
+                    )
+
+    # Create and add stress model
+    stressmodel = getattr(ps.stressmodels, ts["stressmodel"])
+    ts.pop("stressmodel")
+    if "rfunc" in ts.keys():
+        rfunc_kwargs = {}
+        if "rfunc_kwargs" in ts:
+            rfunc_kwargs = ts.pop("rfunc_kwargs")
+        ts["rfunc"] = getattr(ps.rfunc, ts["rfunc"])(**rfunc_kwargs)
+    if "recharge" in ts.keys():
+        recharge_kwargs = {}
+        if "recharge_kwargs" in ts:
+            recharge_kwargs = ts.pop("recharge_kwargs")
+        ts["recharge"] = getattr(ps.recharge, ts["recharge"])(**recharge_kwargs)
+
+    metadata = []
+    settings = []
+
+    # Unpack the stress time series
+    if "stress" in ts.keys():
+        for i, stress in enumerate(ts["stress"]):
+            series, meta, setting = _unpack_series(**stress)
+            ts["stress"][i] = series
+            metadata.append(meta)
+            settings.append(setting)
+
+    if "prec" in ts.keys():
+        series, meta, setting = _unpack_series(**ts["prec"])
+        ts["prec"] = series
+        metadata.append(meta)
+        settings.append(setting)
+
+    if "evap" in ts.keys():
+        series, meta, setting = _unpack_series(**ts["evap"])
+        ts["evap"] = series
+        metadata.append(meta)
+        settings.append(setting)
+
+    if "temp" in ts.keys() and ts["temp"] is not None:
+        series, meta, setting = _unpack_series(**ts["temp"])
+        ts["temp"] = series
+        metadata.append(meta)
+        settings.append(setting)
+
+    if metadata:
+        ts["metadata"] = metadata if len(metadata) > 1 else metadata[0]
+    if settings:
+        ts["settings"] = settings if len(settings) > 1 else settings[0]
+
+    sm = stressmodel(**ts)
+    return sm
+
+
+def _unpack_series(**kwargs):
```

Comment:
why not just accept dictionary? instead of unpacking and repacking using ** and  **kwargs?

Reply:
I will change this to a dict. I was initially thinking to change the export for stressmodels to include the settings and metadata there that is way I used **kwargs. But now I refrained from that so it can be a dict.